packages/go_router/lib/src/delegate.dart

```diff
@@ -14,6 +14,13 @@ import 'match.dart';
 import 'misc/errors.dart';
 import 'route.dart';
 
+/// Signature for a predicate function that determines whether a route should be
+/// popped or not.
+typedef PopUntilPredicate = bool Function(
+  RouteMatchBase,
```

Comment:
RouteMatchBase is currently used internal only, we usually use GoRouterState for public facing API

Reply:
Sure ! I changed it in [refactor: Use GoRouterState instead of RouteMatchBase](https://github.com/flutter/packages/pull/6306/commits/ff97a412a358795ad144b3d96b74ce5c48f53eda)